### PR TITLE
Honour "srid" option in mapnik layer

### LINKED ISSUE #528: MapConfig's mapnik layer SRID not taken into account
It looks like the layer.options.srid is not being added to extra_ds_opts passed to grainstore, when getting a new renderer. This in turn prevents the MapConfig specification for being supported in that reguard.
See https://github.com/CartoDB/Windshaft/blob/master/doc/MapConfig-1.5.0.md#21-mapnik-layers-options

### PR DESCRIPTION
**Pull request by strk**
Fixes #528
Requires an update in grainstore too:
https://github.com/CartoDB/grainstore/pull/126

### CODE REVIEW

**Comment by strk** [comment]
@pramsey can you help getting this in ?

**Comment by strk** [comment]
@jsanz ping

**Comment by strk** [comment]
The grainstore part of this was merged and is available as grainstore 1.6.0 - do you want me to update this PR to include requirement of grainstore 1.6.0 ?

**Comment by strk** [comment]
Well, I've done that, and rebased against master

File: lib/windshaft/renderers/mapnik/factory.js [hunk 1]
```diff
@@ -302,7 +302,13 @@ MapnikFactory.prototype.mapConfigToMMLBuilderConfig = function(mapConfig, queryR
         if ( lyropt.hasOwnProperty('raster_band') ) {
             extra_opt.band = lyropt.raster_band;
         }
-        options.datasource_extend.push(mapConfig.getLayerDatasource(layerIndex));
+
+        var datasource_extend = mapConfig.getLayerDatasource(layerIndex) || {};
```
Comment (rochoa) [inline lib/windshaft/renderers/mapnik/factory.js]: After thinking about this, wouldn't be better if the `mapConfig.getLayerDatasource` returns also the `srid` param?

**Comment by strk** [comment]
Yes, sounds a good idea
But I'm out of funds for this project, so feel free to take it over.

**Comment by rochoa** [comment]
Superseded by #534.

**Pull request closed without merging**